Automattic/wp-calypso
# Stepper auth: add authentication capability without leaving Stepper

## Proposed Changes

~~This makes a thin wrapper around `SignupFormSocialFirst` component and render it in Stepper if the current step requires auth and the user is logged out. It's much more seamless to log the user in without a few redirects back and forth.~~

~~I'm even considering replacing the `WpcomLoginForm` with a fetch request and getting rid of the redirection completely.~~

~~This also moves the `onboarding` flow to use the new User step added in Stepper. This is a proof of concept only. We will certainly need to consider how the tracking of this should look.~~

This adds a `user` Step in Stepper. It also makes Stepper automatically inject this step when any of the steps require authentication. 

**This `onboarding` flow is not in production and is  just a playground for our changes.** 

## Why are these changes being made?

To allow Stepper to seamlessly offer user signup as part of the flow.

## Testing Instructions

This should be tested locally.

1. Mock WordPress.com by running `yarn start-mock-wordpress-com`.
2. Go to https://wordpress.com/setup/onboarding.
3. Signup using different social services and email.
4. All should work as expected. 

## Regression testing
1. Go to /start.
2. Sign up using email. 
3. Should work.
